src/libstd/sys/unix/fd.rs

```diff
@@ -223,50 +222,9 @@ impl FileDesc {
     pub fn duplicate(&self) -> io::Result<FileDesc> {
         // We want to atomically duplicate this file descriptor and set the
         // CLOEXEC flag, and currently that's done via F_DUPFD_CLOEXEC. This
-        // flag, however, isn't supported on older Linux kernels (earlier than
-        // 2.6.24).
-        //
-        // To detect this and ensure that CLOEXEC is still set, we
-        // follow a strategy similar to musl [1] where if passing
-        // F_DUPFD_CLOEXEC causes `fcntl` to return EINVAL it means it's not
-        // supported (the third parameter, 0, is always valid), so we stop
-        // trying that.
-        //
-        // Also note that Android doesn't have F_DUPFD_CLOEXEC, but get it to
```

Comment:
Is this note about android not relevant anymore?

Reply:
I think at the time that flag wasn't in Android's headers, but the implementation details for `fcntl` are all handled in the kernel anyway. AFAICT 2.6.24 is older than any Android kernel that ever shipped.

The `libc` crate moved the Linux definition of the flag to include Android in rust-lang/libc@b43118cb981918a2d8227d59cba9d21ee38af899.